src/main/io/osd.c

```diff
@@ -646,13 +653,34 @@ static bool osdDrawSingleElement(uint8_t item)
             pitchAngle = ((pitchAngle * 25) / AH_MAX_PITCH) - 41; // 41 = 4 * 9 + 5
 
             for (int x = -4; x <= 4; x++) {
-                // clear the y area before writing the new horizon character
-                for (int y = 0; y <= 8; y++) {
-                    displayWriteChar(osdDisplayPort, elemPosX + x, elemPosY + y, SYM_BLANK);
-                }
+                // Don't clear the whole area to save some time. Instead, clear
+                // only the positions we previously wrote iff we're writing
+                // at a different Y coordinate. The video buffer will take care
+                // of ignoring the write if we're writing the same character
+                // at the same Y coordinate.
+                //
+                // Note that this implementation leaves an untouched character
+                // in the bottom center of the indicator, which allows positioning
+                // the home directorion indicator there.
                 const int y = (-rollAngle * x) / 64 - pitchAngle;
+                int wx = x + 4; // map the -4 to the 1st element in the writtenY array
+                int pwy = writtenY[wx]; // previously written Y at this X value
```

Comment:
Why using a new variable here? Compiler (usually) is smart enough to optimize array dereferencing.

Reply:
I just put it in a variable for clarity (same as `ch` a few lines down). I haven't checked the assembly, but I'd expect the compiler to generate the same code.

Reply:
Just checked, exactly the same assembly.